punishments.js

```diff
@@ -871,9 +942,15 @@ Punishments.checkIp = function (user, connection) {
 	let punishment = Punishments.ipSearch(ip);
 
 	if (punishment) {
-		user.locked = punishment[1];
-		if (punishment[0] === 'NAMELOCK') {
-			user.namelocked = punishment[1];
+		if (Punishments.sharedIps.has(user.latestIp)) {
```

Comment:
Rather, shared (and proxy) IPs should be removed from `Punishments.ips` as they're discovered to be shared.